process.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"io"
+
+	"github.com/urfave/cli"
+)
+
+// == Direct input processing ========================================
+
+func processInputFiles(c *cli.Context, g *Gomplate) error {
+	inputs, err := readInputs(c.String("in"), c.StringSlice("file"))
+	if err != nil {
+		return err
+	}
+
+	outputs := c.StringSlice("out")
+	if len(outputs) == 0 {
+		outputs = []string{"-"}
+	}
+
+	for n, input := range inputs {
+		if err := renderTemplate(g, input, outputs[n]); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// == Recursive input dir processing ======================================
+
+func processInputDir(c *cli.Context, g *Gomplate) error {
+	inputDir := c.String("input-dir")
+	outDir := c.String("output-dir")
+	if _, err := assertDir(outDir); err != nil {
```

Comment:
Hrm... 🤔 I think I'd rather create `outDir` if it doesn't exist. `gomplate` is usually used in scripts and typos would be caught pretty quickly, by virtue of the files not existing in the right location. Besides, `--out` will create the output if it doesn't exist, so it makes sense (to me anyway) that `--out-dir` would create the directory too.

Also it makes the code simpler, which is a good thing 😉